repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/AuthPolicyValidator.java

```diff
@@ -307,8 +307,10 @@ public void validate(AtlasEntity policy, AtlasEntity existingPolicy,
             //only allow argo & backend
             if (!RequestContext.get().isSkipAuthorizationCheck()) {
                 String userName = RequestContext.getCurrentUser();
-                validateOperation (!ARGO_SERVICE_USER_NAME.equals(userName) && !BACKEND_SERVICE_USER_NAME.equals(userName),
-                        "Create/Update AuthPolicy with policyCategory other than persona, purpose and datamesh");
+                validateOperation (!ARGO_SERVICE_USER_NAME.equals(userName) && 
```

Comment:
In what scenario/s atlan-governance-workflows supposed to create new policies?
What will be policyCategory & policySubCategory for those policies?

Reply:
To grant sample data read access to a given user, we create a headless AuthPolicy for the requester and add resources to the policy on demand. The AuthPolicy looks something like this:
```
{
  "typeName": "AuthPolicy",
  "attributes": {
    "name": "<policyName>",
    "qualifiedName": "<policyName>",
    "attributes": {
      "policyCategory": "workflows",
      "policyServiceName": "heka",
      "policyType": "allow",
      "policyUsers": ["<requestor>"],
      "policyResourceCategory": "ENTITY",
      "policyResources": <policyResources>,
      "policyActions": ["select"]
    }
  }
}
```

Reply:
1. How to generate name & qualifiedNames?


2. Are these policies re-usable? 
If yes, then how?
- User based : User asks select access on couple of assets OR 
- Resource based: Different users are requesting select access on same asset


3. Since you are not specifying policySubCategory, I hope you have tested data policies with Heka as I fear might throw NPE while refreshing policy cache resulting Data policies created via WF not catering access-control

Reply:
1. The policy name and QN is in the format: "DATA_ACCESS_POLICY/" + requestorUsername
2. Yes these policies are reusable and every user has a single policy (user based), we continue to add resources to it on demand.
3. We have tested these AuthPolicies and they get the job done, but you suggest we can also add an appropriate subCategory.

Reply:
Anyway `policySubCategory` is not a mandatory field, as you have tested I think it is fine to continue without it